templates/core/terraform/appgateway/appgateway.tf

```diff
@@ -0,0 +1,61 @@
+resource "azurerm_public_ip" "appgwpip" {
+  name                  = "pip-agw-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name   = var.resource_group_name
+  location              = var.location
+  allocation_method     = "Static"
+  sku                   = "Standard"
+}
+
+resource "azurerm_application_gateway" "agw" {
+  name                = "agw-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+
+  sku {
+    name     = "Standard_v2"
+    tier     = "Standard_v2"
+    capacity = 2
+  }
+
+  gateway_ip_configuration {
+    name      = "gateway-ip-configuration"
+    subnet_id = var.app_gw_subnet
+  }
+
+  frontend_port {
+    name = local.frontend_port_name
+    port = 80
```

Comment:
Need to create an issue for this - shouldn't be exposing port 80. Should be 443, but will need a certificate.

Reply:
Agree, right now it's base-configured to 80 but not hooked up to anything (so nothing is exposed). This is because if we base-configure to 443, we do need to have a certificate first. Since we don't have a process on acquiring a certificate, we need to have a separate story on certificate provisioning for app gateway @christoferlof if you can add, and then resolve this one.

Reply:
Please see #51

Reply:
The way I've don't this, is to create the listener with a self-signed cert `CN=temp` which allows you to configure the routing correctly. You then replace the cert with `certbot`, or whatever you need to use. I have an example the uses `certbot` I can share.